src/test/kotlin/org/rust/ide/typing/RsBraceMatcherTest.kt

```diff
@@ -8,49 +8,139 @@ package org.rust.ide.typing
 import com.intellij.codeInsight.highlighting.BraceMatchingUtil
 import com.intellij.codeInsight.highlighting.BraceMatchingUtil.getMatchedBraceOffset
 import com.intellij.openapi.editor.ex.EditorEx
+import org.intellij.lang.annotations.Language
 import org.rust.RsTestBase
-import org.rust.lang.RsFileType
 
 class RsBraceMatcherTest : RsTestBase() {
     fun `test don't pair parenthesis before identifier`() = doTest(
-        "fn main() { let _ = <caret>typing }",
+        "fn main() { let _ = /*caret*/typing; }",
         '(',
-        "fn main() { let _ = (<caret>typing }"
+        "fn main() { let _ = (/*caret*/typing; }"
+    )
+
+    fun `test don't pair parenthesis before string literal`() = doTest(
+        """fn main() { let _ = /*caret*/"foo"; }""",
+        '(',
+        """fn main() { let _ = (/*caret*/"foo"; }"""
+    )
+
+    fun `test don't pair parenthesis before integral literal`() = doTest(
+        "fn main() { let _ = /*caret*/1; }",
+        '(',
+        "fn main() { let _ = (/*caret*/1; }"
+    )
+
+    fun `test don't pair parenthesis before ref operator`() = doTest(
+        "fn main() { let _ = /*caret*/&foo; }",
+        '(',
+        "fn main() { let _ = (/*caret*/&foo; }"
+    )
+
+    fun `test don't pair parenthesis before dereference operator`() = doTest(
+        "fn main() { let _ = /*caret*/*foo.bar; }",
+        '(',
+        "fn main() { let _ = (/*caret*/*foo.bar; }"
```

Comment:
We can't discriminate in the brace matcher a dereference from a non-formatted binary multiplication. I believe we should match parentheses (and brackets, and braces) here.